sql/moz-fx-data-shared-prod/telemetry_derived/feature_usage_v2/script.sql

```diff
@@ -0,0 +1,760 @@
+-- This script generates the feature usage data for a specific submission_date.
+-- Coercions of values is done automatically based on the type and min_version provided
+-- in the temporary `version_metadata` table. In particular, all nulls for numerical measures
+-- are coalesced to 0 in cases where the client was on an eligible browser version (i.e. a
+-- version where the underlying telemetry was implemented); otherwise leave the measure as null
+-- for ineligible versions.
+-- For adding a new metric:
+-- (1) Add metadata to `version_metadata` for the new metric. Metadata is specified as a list of
+--   STRUCTs where `metric` specifies the column name for the new feature metric in the result table,
+--   `min_version` specifies the minimum eligible browser version and `type` the data type of the
+--   computed metric value.
+-- (2) Update `feature_usages` and specify the feature metric
+-- (3) Update the schema.yaml file and add the new field to the schema
+-- (4) Run `./bqetl query schema deploy telemetry_derived.feature_usage_v2 --force` to update the destination
+--   table schema.
```

Comment:
So I think I could personally manage this workflow, but my guess is that for broader usability people would find updating a yaml file more familiar. opinions from @SuYoungHong and @godelstheory welcome on this point.

I think the question is, if only data folks are going to be making updates here, then maybe this kind of workflow is OK (I'm talking purely based on the ergonomics). If the table is getting so much use that PMs want to be able to add their own metrics, then this workflow is likely going to be too involved. But, we don't have any evidence for that yet. 

That said, maybe there are technical benefits to taking the extra time to yaml-fy that I don't understand. For example, if it sets a precedent or creates res-usable code that can be used for similar tables in the future, then I would be all for that.

Reply:
Also, if I'm reading this correctly, if I go through this workflow and add a metric, then it will just be present for all dates after the first run of the ETL, correct? Its not going to automatically backfill previous dates, or will it?

Reply:
I agree, having YAML files to configure the metrics DS is interested in probably would it make easier for DS to make updates. I'll see if there is a good generic path forward to support YAML files in bigquery-etl since we had some related use cases already. 

As for backfilling, previous dates are not going to automatically backfill in general. 
There are 2 recommended ways for performing a backfill:
* via `bqetl` command line tool: run `./bqetl query backfill telemetry_derived.feature_usage_v2 --start_date=<start> --end_date=<end>` locally
* or via the [Airflow Admin Backfill UI](https://workflow.telemetry.mozilla.org/admin/backfill/) (sometimes it's a bit flaky)